File: pykubachu-admin/pykuybachu-admin.py
#!/usr/bin/env python3

import os
import sys
from kubernetes import client, config

NAME = os.environ.get('NAME', 'pykubachu')
NAMESPACE = os.environ.get('NAMESPACE', 'chipy')

def update_pykubachu(color):
    config.load_kube_config()
    extensions_v1beta1 = client.ExtensionsV1beta1Api()

    deployment = extensions_v1beta1.read_namespaced_deployment(
        name=NAME,
        namespace=NAMESPACE)

    deployment.spec.template.spec.containers[0].env[0].value = color
    api_response = extensions_v1beta1.patch_namespaced_deployment(
        name=NAME,
        namespace=NAMESPACE,
        body=deployment)

    print("Deployment updated. status='%s'" % str(api_response.status))

if __name__ == '__main__':
    args = sys.argv
    if len(args) != 2:
        print('Usage: ./pykuybachu-admin.py <color>')
        sys.exit(1)

    color = args[1]
    update_pykubachu(color)

File: ruby/pykubachu-admin/pykuybachu-admin.rb
#!/usr/bin/env ruby

require 'kubeclient'

tag = ARGV[0]

config = Kubeclient::Config.read(ENV['KUBECONFIG'])
context = config.context
ssl_options = context.ssl_options
auth_options = context.auth_options

client = Kubeclient::Client.new(
    context.api_endpoint, 'v1',
    ssl_options: ssl_options, auth_options: auth_options
)

for pod in client.get_pods do
    c1 = pod.spec.containers[0]
    if c1.name == "pykubachu"
        c1.image = "bloodorangeio/pykubachu:#{tag}"
        client.update_pod(pod)
    end
end
